src/app.rs

```diff
@@ -322,10 +336,69 @@ pub struct Config<'a> {
     pub term_width: usize,
     pub files: Vec<Option<&'a str>>,
     pub theme: Option<&'a str>,
+    pub line_range: Option<LineRange>,
 }
 
 fn is_truecolor_terminal() -> bool {
     env::var("COLORTERM")
         .map(|colorterm| colorterm == "truecolor" || colorterm == "24bit")
         .unwrap_or(false)
 }
+
+pub struct LineRange {
+    pub lower: usize,
+    pub upper: usize,
+}
+
+impl LineRange {
+    pub fn from(range_raw: &str) -> LineRange {
+        LineRange::parse_range(range_raw).unwrap_or(LineRange::new())
+    }
+
+    pub fn new() -> LineRange {
+        LineRange{
+            lower: usize::min_value(),
+            upper: usize::max_value(),
+        }
+    }
+
+    pub fn parse_range(range_raw: &str) -> Result<LineRange> {
+        let mut new_range = LineRange::new();
+
+        if range_raw.bytes().nth(0).ok_or("No first byte")? == b':' {
+            new_range.upper = range_raw[1..].parse()?;
+            return Ok(new_range);
+        } else if range_raw.bytes().last().ok_or("No last byte")? == b':' {
+            new_range.lower = range_raw[..range_raw.len() - 1].parse()?;
+            return Ok(new_range);
+        }
+
+        let line_numbers: Vec<&str> = range_raw.split(':').collect();
+        if line_numbers.len() == 2 {
```

Comment:
I think we should have an `else` branch here which returns a error ("expected single ':' character").

Reply:
This might be a silly question but how do I go about creating an error? When I try to return
```Err("expected single....")``` I get a mismatched types error: ```expected struct `errors::Error`, found reference```

Do I need to implement a new error type or something like that?

Reply:
You should be able to use `Err("expected single....".into())`.